### ISSUE TITLE
Add `.only` and `.skip` modifiers

### ISSUE THREAD

**Issue by mrazauskas** [wellwelwel/poku#722]
Please add `.only` and `.skip` modifiers.

This would help a lot while debugging or developing new features.

As a work around, currently one can comment out failing or not yet working tests to skip them. But this can create unused imports and these can be removed by formatters automatically. I just try to explain that comment / uncomment is not the same as `.skip`.

**Comment by wellwelwel** [comment]
Hey @mrazauskas 🙋🏻‍♂️

About `skip`, I'm developing this feature allowing conditions to enable it or not. For example:

```ts
test.skip(isWindows, ('My Test') => {
  // ...
});
```

- Still thinking about the best way to organize the params.

On the other hand, **Poku** doesn't map the tests before running them, it just allows the standard JavaScript flow. I need to elaborate an approach to make it possible to use `only` without compromising this principle.

**Comment by mrazauskas** [comment]
Regarding the flow. I think, it is enough to define conditional tests:

```js
test("conditional test", () => {
  if (isWindows) {
    return;
  }

  // etc.
});
```

One can print a message, or not. This works.

---

What if importing `only()` (as an alias for `describe()` and `test()`) would have some side effect that would make it work? Just an idea. I don’t know how to implement this. It feels like side effect thing.

**Comment by wellwelwel** [comment]
> What if importing `only()` (as an alias for `describe()` and `test()`) would have some side effect that would make it work?

I'll play with that.

**Comment by wellwelwel** [comment]
All ideas when mapping tests always come back to this problem:

```ts
describe(() => {
  console.log('Start');

  it('Test 1', () => {
    // ...
  });

  it.only('Test 2', () => {
    // ...
  });

  it('Test 3', () => {
    // ...
  });

  console.log('End');
});
```

Output:

```
Start
End
Test 2
```

Although it should be:

```
Start
Test 2
End
```

---

I'm thinking something like:

```ts
import { describe, test } from 'poku';

describe.modifiers.only.enable();

test('', () => {});

test.only('', () => {}); // ✅

test('', () => {});
```

- That's way, there is no `organizeImports` conflicts.
- It's important to note that if there is no ".only", nothing will be performed (maybe force an error in this case).
- The same idea when using `.only` without explicitly defining the modifier.

**Comment by wellwelwel** [comment]
> Maybe:
> 
> ```ts
> describe.modifiers.only.enable();
> test.modifiers.only.enable();
> it.modifiers.only.enable();
> ```

It's also interesting to note that my idea is not to map the tests if these option are passed, but rather to internally ignore the tests that don't come from `only`.

**Comment by mrazauskas** [comment]
Hm.. It is easy to forget to remove that line. Also it looks verbose and not really clear why `only` must be specified separately for `describe()` and `test()`?

What about something simple:

```js
import { modifiers } from 'poku';

modifiers({ only: true });
```

In the other hand, this feature could be only available via test runner. For instance, Node.js test runner has the [`--test-only`](https://nodejs.org/api/cli.html#--test-only) flag. Never tried it, but it looks like [Bun has something](https://bun.sh/docs/test/writing#test-only) similar: `bun test --only`.

**Comment by mrazauskas** [comment]
By the way, take a look at this note: https://bun.sh/blog/bun-v1.1.26#test-only-skips-all-other-tests

I don’t know how that is implemented, but seems like DX got improved in the latest version of Bun.

**Comment by wellwelwel** [comment]
> ```js
> import { modifiers } from 'poku';
> 
> modifiers({ only: true });
> ```

That way, both test runner and direct tests would work, but we'd have the same problem by forgetting to remove that line line.

---

> For instance, Node.js test runner has the [`--test-only`](https://nodejs.org/api/cli.html#--test-only) flag. Never tried it, but it looks like [Bun has something](https://bun.sh/docs/test/writing#test-only) similar: `bun test --only`.

This approach would only work with the test runner, but despite this, I feel that this is a suitable alternative, even if it doesn't work with the direct test:

- No unnecessary code changes 
- No need to change imports
- A temp flag like `--only` makes sense for a debugging process
- It can be used without changing the script, e.g: `npm run test -- --only`

Also, it would be easy to warn runs without the test runner that `.only` will be run as a common test (no modifiers).

I'll try to elaborate on that idea first.

---

> I don’t know how that is implemented, but seems like DX got improved in the latest version of Bun.

This is done at **Zig** level: https://github.com/oven-sh/bun/commit/781998cf003ae4b61adc864c908990825112e037 (from our side, it would be like interpreting the file before executing it).

**Comment by wellwelwel** [comment]
> _Just an update:_

The `skip` is available in `v2.6.0`.

About the `.only`:

It was quite simple to solve the question of whether to allow `.only` with the test runner or without it.

- I opted to use the `--only` flag.
- If no flags are passed, but there are `.only` tests, I opted to cause a failure in the test in question.

> Feel free to give your opinion 🙋🏻‍♂️

**Comment by mrazauskas** [comment]
Sounds good to me. Especially the idea about throwing an error, because one can commit some `.only` tests accidentally, hence getting an error in CI is great thing!
